mitmproxy/addons/command_history.py

```diff
@@ -0,0 +1,113 @@
+import collections
+import os
+import typing
+import atexit
+
+from mitmproxy import command
+from mitmproxy import ctx
+
+
+class CommandHistory:
+    def __init__(self, size: int = 300) -> None:
+        self.saved_commands: typing.Deque[str] = collections.deque(maxlen=size)
+
+        self.filtered_commands: typing.Deque[str] = collections.deque()
+        self.current_index: int = -1
+        self.filter_str: str = ''
+
+        _command_history_dir = os.path.expanduser(ctx.options.confdir)
+        if not os.path.exists(_command_history_dir):
+            os.makedirs(_command_history_dir)
+
+        _command_history_path = os.path.join(_command_history_dir, 'command_history')
+        _history_lines: typing.List[str] = []
+        if os.path.exists(_command_history_path):
+            _history_lines = open(_command_history_path, 'r').readlines()
```

Comment:
Let's make sure that we always close files after reading:

1) Use `with open(...):` .... so that the file pointer is automatically closed.
2) Use pathlib and https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text

Reply:
I thought that the way I wrote it would cause the file to be closed after the readlines() call. My bad I guess. Will change it to use `with`. Any reason to use pathlib instead?

Reply:
This is what pathlib is doing internally: https://github.com/python/cpython/blob/b9e5547f5814962964c4a5bd5cd36a2af8fbf974/Lib/pathlib.py#L1220-L1225
So I guess the answer is no. Personal preference. 😃

Reply:
I changed it to use `with open` then. Should be good now